de-de/c-de.html.markdown

```diff
@@ -0,0 +1,880 @@
+    // Beachte, dass die Länge des Arrays nicht explizit definiert werden muss, 
+    // wenn er auf derselben Linie initialisiert wird.
+    // Folgende Deklaration ist gleichwertig:
+    char mein_array[] = {0};
+    // Allerdings muss die Länge des Arrays dann zur Laufzeit ausgewertet werden:
+    size_t mein_array_size = sizeof(mein_array) / sizeof(mein_array[0]);
+    // WARNUNG: Wenn dieser Ansatz gewählt wird, muss man sicherstellen, dass die
+    // Grösse des Arrays ermittelt werden *bevor* dieser einer Funktion als
+    // Argument weitergegeben wird (siehe Diskussion weiter unten), weil Arrays
+    // einer Funktion nur als Zeiger übergeben werden. => Das obere Statement
+    // würde innerhalb einer Funktion ein falsches Resultat liefern.
+
+    // Das Indexieren eines Arrays funktioniert wie in anderen Sprache - resp.
+    // in anderen Sprachen funktioniert es gleich wie in C.
+    mein_array[0]; // => 0
+    
+    // Arrays sind veränderbar; es ist nur Arbeitsspeicher!
+    mein_array[1] = 2;
+    printf("%d\n", mein_array[1]); // => 2
+
+    // In C99 (und als optionales Feature in C11) können Arrays mit variabler
+    // Länge deklariert werden. Die Grösse eines solchen Array muss eine Konstante
+    // zur Kompilierzeit sein.
+    printf("Geben Sie die Arraygrösse an: "); //Frag den Benutzer nach
+                                              // der Arraygrösse
+    int array_size;
+    fcsanf(stdin, "%d", &array_size);
+    int var_length_array[array_size]; // deklariere Array mit variabler Länge
+    printf("sizeof array =%zu\n", sizeof var_length_array);
+
+    // Zum Beispiel:
+    // > Geben Sie die Arraygrösse an: 10
+    // > sizeof array = 40
+
+    // Strings sind lediglich Arrays von `chars`, welche mit einem Null-Byte
+    // (0x00) beendet werden. In Strings wird das Nullbyte durch das Zeichen \0
+    // repräsentiert. Wir müssen das Null-Byte nicht angeben in String-Literalen;
+    // der Compiler fügt es am Ende des Array automatisch hinzu.
+    char ein_string[20] = "Das ist ein String";
+    printf("%s\n", ein_string); // %s formattiert einen String
+
+    printf("%d\n", ein_string[18]); // => 0
+    // Hier ist das Byte #19 0 (wie auch Byte #20)
+
+    // Wenn wir Zeichen zwischen einfachen Anführungszeichen haben, ist es ein
+    // Zeichenliteral vom Typ int und *nicht* char. (aus historischen Gründen)
+    int cha = 'a'; // Ok
+    char chb = 'a'; // auch ok (implizite Umwandlung von int zu char)
+
+    // Mehrdimensionale Arrays:
+    int multi_array[2][5] = {
+        {1,2,3,4,5},
+        {6,7,8,9,0}
+    };
+    // Auf Elemente zugreifen:
+    int array_int = multi_array[0][2]; // => 3
+
+    ////////////////////////////////////////////////
+    // Operatoren
+    ////////////////////////////////////////////////
+    
+    // Kurzschreibweise für mehrere Deklarationen
+    int i1 = 1, i2 = 2;
+    flaot f1 = 1.0, f2 = 2.0;
+
+    int b,c;
+    b = c = 0;
+
+    // Arithmetik ist unkompliziert
```

Comment:
I believe basic arithmetic and logical operators should be closer to the beginning of the file, before arrays etc

The order in the Python article is quite healthy: https://learnxinyminutes.com/docs/python3/

Reply:
Sounds good to me. I didn't think about that.

In that case it should also be changed in the English tutorial, isn't it?